File: socialuni-sdk/src/main/java/com/socialuni/social/sdk/feignAPI/openData/SocialuniLocationAPI.java
package com.socialuni.social.sdk.feignAPI.openData;

import com.socialuni.social.common.api.model.ResultRO;
import com.socialuni.social.user.sdk.model.QO.LocationQueryQO;
import com.socialuni.social.user.sdk.model.RO.LocationQueryRO;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

import javax.validation.Valid;

/**
 * @author qinkaiyuan
 * @date 2018-11-18 20:45
 */

//@RequestMapping("location")
@FeignClient(name = "location", url = "${socialuni.central-server-url:https://api.socialuni.cn}", path = "socialuni/location")
@Tag(name = "开放数据/腾讯云内容")
public interface SocialuniLocationAPI {
    @PostMapping("queryLocation")
    ResultRO<LocationQueryRO> queryLocation(@RequestBody @Valid LocationQueryQO queryQO);
}

File: socialuni-js/packages/socialuni-community-api/src/api/SocialuniLocationAPI.ts
import socialuniCommunityRequest from "../request/socialuniCommunityRequest";
import LocationQueryQO from "socialuni/src/model/location/LocationQueryQO";
import {QQMapResult} from "socialuni/src/model/location/QQMapResult";
import ResultRO from "socialuni/src/model/social/ResultRO";


export default class SocialuniLocationAPI {
    static queryLocation(location: LocationQueryQO = new LocationQueryQO()): Promise<ResultRO<QQMapResult>> {
        //小程序，且注册了qq的地图key
        return socialuniCommunityRequest.post<QQMapResult>('socialuni/location/queryLocation', location) as any
    }
}
